### PR TITLE
feat: Support for Escoria and Game migrations

### PR DESCRIPTION
**Pull request by dploeger**
This integrates an ESCMigrationManager which can run migration scripts to modify savegames through a series of semver-based migration files when the game is loaded.

fixes godot-escoria/escoria-issues#18

### CODE REVIEW

File: addons/escoria-core/game/core-scripts/migrations/esc_migration.gd [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Base class for all migration version scripts. Extending scripts should be
+# named like the version they migrate the savegame to. (e.g. 1.0.0.gd, 1.0.1.gd)
+extends Node
```
Comment (StraToN) [inline addons/escoria-core/game/core-scripts/migrations/esc_migration.gd:3]: Why does ESCMigration needs to be a Node? Does it need to be added to a scene, or could it be instanced as an Object instead?

File: addons/escoria-core/game/core-scripts/migrations/esc_migration_manager.gd [hunk 2]
```diff
@@ -0,0 +1,172 @@
+# Class that handles migrations between different game or escoria versions
+extends Node
```
Comment (StraToN) [inline addons/escoria-core/game/core-scripts/migrations/esc_migration_manager.gd:2]: Same question here: does it need to be added to a scene, or could it be instanced as an Object instead?

**Review by StraToN — CHANGES_REQUESTED** [review]

File: addons/escoria-core/game/core-scripts/migrations/esc_migration.gd (hunk 1 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/migrations/esc_migration.gd:3]: Yes, you're right. Probably can go with object. :)

File: addons/escoria-core/game/core-scripts/migrations/esc_migration_manager.gd (hunk 2 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/migrations/esc_migration_manager.gd]: Done

File: addons/escoria-core/game/core-scripts/migrations/esc_migration.gd (hunk 1 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/migrations/esc_migration.gd]: Done

**Review by StraToN — APPROVED** [review]

**Comment by StraToN** [comment]
Seems all good to me! Fits exactly the requirement, thank you so much!

**Pull request merged**